base/set.jl

```diff
@@ -190,6 +190,75 @@ function unique(f::Callable, C)
     out
 end
 
+# If A is not sorted, then we will need to keep track of all of the elements that
+# we have seen so far.
+function _unique!(A)
+    seen = Set{eltype(A)}()
+    idxs = eachindex(A)
+    i = state = start(idxs)
+    for x in A
+        if x ∉ seen
+            push!(seen, x)
+            i, state = next(idxs, state)
+            A[i] = x
+        end
+    end
+    resize!(A, i - first(idxs) + 1)
+end
+
+# If A is sorted, then we only need to keep track of one element and add that to A
+# every time that we see a new element.
+function _sortedunique!(A)
+    idxs = eachindex(A)
+    y = first(A)
+    state = start(idxs)
+    j, state = next(idxs, state)
+    for i in idxs
+        x = A[i]
+        if !isequal(x, y)
+            j, state = next(idxs, state)
+            y = A[j] = x
+        end
+    end
+    resize!(A, j - first(idxs) + 1)
+end
+
+"""
+    unique!(A::AbstractVector)
+
+Remove duplicate items as determined by [`isequal`](@ref), then return the modified `A`.
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A);
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+```
+"""
+function unique!(A::AbstractVector{<:Union{Real, AbstractString, Symbol}})
```

Comment:
Strictly speaking, this should be `::Union{AbstractVector{<:Real}, AbstractVector{<:AbstractString}, AbstractVector{<:Symbol}}`, since `issorted([1, "a"])` throws an error.